src/configuration.js

```diff
@@ -0,0 +1,252 @@
+ * `actorEventsWsUrl` | `ACTOR_EVENTS_WS_URL` | -
+ * `inputKey` | `INPUT_KEY` | `'INPUT'`
+ * `actorId` | `ACTOR_ID` | -
+ * `apiBaseUrl` | `API_BASE_URL` | `'https://api.apify.com/v2'`
+ * `isAtHome` | `IS_AT_HOME` | -
+ * `actorRunId` | `ACTOR_RUN_ID` | -
+ * `actorTaskId` | `ACTOR_TASK_ID` | -
+ * `containerPort` | `CONTAINER_PORT` | `4321`
+ * `containerUrl` | `CONTAINER_URL` | `'http://localhost:4321'`
+ * `userId` | `USER_ID` | -
+ * `proxyHostname` | `PROXY_HOSTNAME` | `'proxy.apify.com'`
+ * `proxyPassword` | `PROXY_PASSWORD` | -
+ * `proxyStatusUrl` | `PROXY_STATUS_URL` | `'http://proxy.apify.com'`
+ * `proxyPort` | `PROXY_PORT` | `8000`
+ * `maxOpenedStorages` | `MAX_OPENED_STORAGES` | `1000`
+ *
+ * ## Not Supported environment variables
+ *
+ * - `MEMORY_MBYTES`
+ * - `HEADLESS`
+ * - `XVFB`
+ * - `CHROME_EXECUTABLE_PATH`
+ */
+export class Configuration {
+    // all env vars also supports `APIFY_` prefix
+    static ENV_MAP = {
+        TOKEN: 'token',
+        LOCAL_STORAGE_DIR: 'localStorageDir',
+        LOCAL_STORAGE_ENABLE_WAL_MODE: 'localStorageEnableWalMode',
+        DEFAULT_DATASET_ID: 'defaultDatasetId',
+        DEFAULT_KEY_VALUE_STORE_ID: 'defaultKeyValueStoreId',
+        DEFAULT_REQUEST_QUEUE_ID: 'defaultRequestQueueId',
+        METAMORPH_AFTER_SLEEP_MILLIS: 'metamorphAfterSleepMillis',
+        PERSIST_STATE_INTERVAL_MILLIS: 'persistStateIntervalMillis',
+        TEST_PERSIST_INTERVAL_MILLIS: 'persistStateIntervalMillis', // for BC, seems to be unused
+        ACTOR_EVENTS_WS_URL: 'actorEventsWsUrl',
+        INPUT_KEY: 'inputKey',
+        ACTOR_ID: 'actorId',
+        API_BASE_URL: 'apiBaseUrl',
+        IS_AT_HOME: 'isAtHome',
+        ACTOR_RUN_ID: 'actorRunId',
+        ACTOR_TASK_ID: 'actorTaskId',
+        CONTAINER_PORT: 'containerPort',
+        CONTAINER_URL: 'containerUrl',
+        USER_ID: 'userId',
+        PROXY_HOSTNAME: 'proxyHostname',
+        PROXY_PASSWORD: 'proxyPassword',
+        PROXY_STATUS_URL: 'proxyStatusUrl',
+        PROXY_PORT: 'proxyPort',
+        MAX_OPENED_STORAGES: 'maxOpenedStorages',
+
+        // not supported, use env vars directly:
+        // MEMORY_MBYTES: 'memoryMbytes',
+        // HEADLESS: 'headless',
+        // XVFB: 'xvfb',
+        // CHROME_EXECUTABLE_PATH: 'chromeExecutablePath',
+    };
+
+    static ENV_MAP_REVERSED = Object.entries(Configuration.ENV_MAP).reduce((obj, [key, value]) => {
+        obj[value] = key;
+        return obj;
+    }, {});
+
+    static BOOLEAN_VARS = ['localStorageEnableWalMode'];
+
+    static INTEGER_VARS = ['proxyPort', 'internalPort', 'memoryMbytes', 'containerPort'];
+
+    static DEFAULTS = {
+        defaultKeyValueStoreId: 'default',
+        defaultDatasetId: 'default',
+        defaultRequestQueueId: 'default',
+        inputKey: 'INPUT',
+        proxyHostname: 'proxy.apify.com',
+        apiBaseUrl: 'https://api.apify.com/v2',
+        proxyStatusUrl: 'http://proxy.apify.com',
+        proxyPort: 8000,
```

Comment:
ok will keep the defaults here but use the values from `apify-shared`

Reply:
just to be sure, i should use `LOCAL_ENV_VARS` for everything except the first 3 items (default storage names)? here is the full object:

```ts
const LOCAL_ENV_VARS = {
  [ENV_VARS.DEFAULT_KEY_VALUE_STORE_ID]: 'default',
  [ENV_VARS.DEFAULT_DATASET_ID]: 'default',
  [ENV_VARS.DEFAULT_REQUEST_QUEUE_ID]: 'default',
  [ENV_VARS.PROXY_HOSTNAME]: 'proxy.apify.com',
  [ENV_VARS.PROXY_PORT]: 8000 .toString(),
  [ENV_VARS.CONTAINER_PORT]: DEFAULT_CONTAINER_PORT.toString(),
  [ENV_VARS.CONTAINER_URL]: `http://localhost:${DEFAULT_CONTAINER_PORT}` // Must match port line above!
};
```

(that way its already done)

Reply:
We must use the values from shared for the 4 keys I highlighted, because those are relevant even when running locally. If you have a bad proxy hostname or a bad API base URL, some parts of the SDK will not work.

Container port and URL do not need to be pulled from shared, because they are only relevant on the platform.

Default storage env vars are also used in the `apify-cli`. When you run `apify run -p` to purge the default storages, it looks at those env vars to find the names of folders to delete and falls back to `LOCAL_ENV_VARS`. So I guess for now, before we redesign the relationship between SDK and CLI, we should also use the default storage names from the `LOCAL_ENV_VARS`

Reply:
Hmm, interesting. I thought that the `APIFY_API_BASE_URL` and `APIFY_PROXY_STATUS_URL` are also provided by `shared`. Apparently the defaults are not there.